File: php/src/Model/TeamPresenceMember.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * One worker presence.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class TeamPresenceMember
{
    public function __construct(
        public readonly string $workerId,
        public readonly string $label,
        /** working | on-break | paused */
        public readonly string $status,
        public readonly ?string $breakStartedAt,
        public readonly ?string $breakReason,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            workerId: (string) ($data['workerId'] ?? ''),
            label: (string) ($data['label'] ?? ''),
            status: (string) ($data['status'] ?? ''),
            breakStartedAt: isset($data['breakStartedAt']) ? (string) $data['breakStartedAt'] : null,
            breakReason: isset($data['breakReason']) ? (string) $data['breakReason'] : null,
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/TeamPresenceMember.java
package io.fivexer.sdk.model;

/**
 * One worker's presence.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class TeamPresenceMember {
    private String workerId;
    private String label;
    private String status;  // working | on-break | paused
    private String breakStartedAt;
    private String breakReason;

    public String getWorkerId() { return workerId; }
    public String getLabel() { return label; }
    public String getStatus() { return status; }
    public String getBreakStartedAt() { return breakStartedAt; }
    public String getBreakReason() { return breakReason; }
}
